api/common/v1alpha1/entitytype.go

```diff
@@ -0,0 +1,11 @@
+package v1alpha1
+
+// EntityType is the type for all the entity types.
+type EntityType string
+
+const (
+	// EntityTypeOrigin is the type for Origin entities.
+	EntityTypeOrigin EntityType = "Origin"
```

Comment:
could we use `origin` and `mirror`? In my mind the constants usually start with lower case letters.

Reply:
The upstream api convention states that the allowed values for the enum fields should be `CamelCase`: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants